### PR TITLE
BUG: Mark CMake variables as advanced

### LINKED ISSUE #4446: Clean up advanced CMake variables
There are some variables that should marked as advanced CMake variables with CMake`s `mark_as_advanced`. A current default configuration + `CMAKE_EXPORT_COMPILE_COMMANDS`:

image

Of these, only `BUILD_EXAMPLES`, `BUILD_TESTING`, `ITK_BUILD_DEFAULT_MODULES`, `BUILD_STATIC_LIBS`, `ITK_WRAP_PYTHON` should not be marked as advanced.

Also, `CMAKE_BUILD_TYPE` should not be advanced.

### PR DESCRIPTION
**Pull request by thewtex**
These are mostly from third party libraries. Add `marked_as_advanced` to
the vendored code.

Keep the standard configuration simple.

Closes #4446

### CODE REVIEW

**Review by phcerdan — APPROVED** [review]

File: Modules/Remote/IOMeshSWC.remote.cmake [hunk 1]
```diff
@@ -49,3 +49,4 @@ itk_fetch_module(
   GIT_REPOSITORY https://github.com/InsightSoftwareConsortium/ITKIOMeshSWC.git
   GIT_TAG 6375295e7802c6a064c90d11f48b3ea238ac8d76
   )
+mark_as_advanced(FORCE Module_IOMeshSWC)
```
Comment (dzenanz) [inline Modules/Remote/IOMeshSWC.remote.cmake:52]: Why mark this as advanced, while most other remote modes are not? And why is `FORCE` needed?

File: Modules/ThirdParty/GoogleTest/src/itkgoogletest/googletest/CMakeLists.txt [hunk 2]
```diff
@@ -337,3 +337,7 @@ if (gtest_build_tests)
   py_test(gtest_xml_output_unittest --no_stacktrace_support)
   py_test(googletest-json-output-unittest --no_stacktrace_support)
 endif()
+
+# Begin ITK additions
+mark_as_advanced(GTEST_HAS_ABSL)
+# End ITK additions
```
Comment (N-Dekker) [inline Modules/ThirdParty/GoogleTest/src/itkgoogletest/googletest/CMakeLists.txt:343]: Would it be possible to move those ITK additions to the _ITK specific_ CMakeLists.txt at "Modules/ThirdParty/GoogleTest", https://github.com/InsightSoftwareConsortium/ITK/blob/ce74ac3181cd4f6adb1d7680f793a74908549e99/Modules/ThirdParty/GoogleTest/CMakeLists.txt ?  If so, I think that would be preferable 🤷

File: Modules/ThirdParty/ZLIB/src/itkzlib-ng/CMakeLists.txt [hunk 3]
```diff
@@ -1307,3 +1307,6 @@ endif()
 
 include(CPack)
 endif() # ITK change
+# Begin ITK additions
+mark_as_advanced(WITH_GTEST ZLIBNG_ENABLE_TESTS)
+# End ITK additions
```
Comment (N-Dekker) [inline Modules/ThirdParty/ZLIB/src/itkzlib-ng/CMakeLists.txt:1312]: Sorry, I have the same question here: would it be possible to move these ITK additions to the ITK specific CMakeLists.txt at "Modules/ThirdParty/ZLIB"?

File: Modules/Remote/IOMeshSWC.remote.cmake (hunk 1 above)
Comment (thewtex) [reply Modules/Remote/IOMeshSWC.remote.cmake:52]: I am not sure why it is needed here.

**Comment by dzenanz** [comment]
Should `ITK_WRAP_*` (e.g. `ITK_WRAP_complex_float`) be advanced?

File: Modules/ThirdParty/GoogleTest/src/itkgoogletest/googletest/CMakeLists.txt (hunk 2 above)
Comment (thewtex) [reply Modules/ThirdParty/GoogleTest/src/itkgoogletest/googletest/CMakeLists.txt:343]: Yes!

**Comment by thewtex** [comment]
> Should ITK_WRAP_* (e.g. ITK_WRAP_complex_float) be advanced?

No, we want the wrapped pixel types and dimension exposed --they are commonly.

**Review by dzenanz — APPROVED** [review]

**Review by hjmjohnson — APPROVED** [review]

File: Modules/ThirdParty/ZLIB/src/itkzlib-ng/CMakeLists.txt (hunk 3 above)
Comment (N-Dekker) [reply Modules/ThirdParty/ZLIB/src/itkzlib-ng/CMakeLists.txt]: Thanks for addressing my comments @thewtex ! 👍

**Pull request merged**